core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -13,86 +13,186 @@
     {
-        return new ConnectorToSqlExpressionTranslator(variableMappings, literalEncoder).translate(expression);
+        return new ConnectorToSqlExpressionTranslator(session, metadata, literalEncoder, variableMappings)
+                .process(expression)
+                .orElseThrow(() -> new UnsupportedOperationException("Expression type not supported: " + expression.getClass().getName()));
     }
 
-    public static Optional<ConnectorExpression> translate(Session session, Expression expression, TypeAnalyzer types, TypeProvider inputTypes)
+    public static Optional<ConnectorExpression> translate(Session session, Expression expression, TypeAnalyzer types, TypeProvider inputTypes, Metadata metadata)
     {
-        return new SqlToConnectorExpressionTranslator(types.getTypes(session, inputTypes, expression))
+        return new SqlToConnectorExpressionTranslator(session, types.getTypes(session, inputTypes, expression), metadata)
                 .process(expression);
     }
 
     private static class ConnectorToSqlExpressionTranslator
     {
-        private final Map<String, Symbol> variableMappings;
+        private final Session session;
+        private final Metadata metadata;
         private final LiteralEncoder literalEncoder;
+        private final Optional<Map<String, Symbol>> variableMappings;
 
-        public ConnectorToSqlExpressionTranslator(Map<String, Symbol> variableMappings, LiteralEncoder literalEncoder)
+        public ConnectorToSqlExpressionTranslator(Session session, Metadata metadata, LiteralEncoder literalEncoder, Optional<Map<String, Symbol>> variableMappings)
         {
-            this.variableMappings = requireNonNull(variableMappings, "variableMappings is null");
+            this.session = requireNonNull(session, "session is null");
+            this.metadata = requireNonNull(metadata, "metadata is null");
             this.literalEncoder = requireNonNull(literalEncoder, "literalEncoder is null");
+            this.variableMappings = requireNonNull(variableMappings, "variableMappings is null");
         }
 
-        public Expression translate(ConnectorExpression expression)
+        public Optional<Expression> process(ConnectorExpression expression)
         {
-            if (expression instanceof Variable) {
-                return variableMappings.get(((Variable) expression).getName()).toSymbolReference();
+            if (expression instanceof Constant) {
+                return Optional.of(literalEncoder.toExpression(((Constant) expression).getValue(), expression.getType()));
+            }
+            else if (expression instanceof FieldDereference) {
+                FieldDereference fieldDereference = (FieldDereference) expression;
+                Optional<Expression> baseOpt = process(fieldDereference.getTarget());
```

Comment:
Don't abbreviate variable names. In fact, the variable is unnecessary, so you can just remove it.

Reply:
Done